src/gui/optionsdialog.cpp

```diff
@@ -230,9 +232,30 @@ OptionsDialog::OptionsDialog(QWidget *parent)
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC)) && !defined(QT_DBUS_LIB)
     m_ui->checkPreventFromSuspend->setDisabled(true);
 #endif
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
```

Comment:
And these too. I would "export" from the os.h file `QWidget* fileAssociationsWidget (QWidget* parent)` and enjoy clean header, would not introduce <Windows.h> contents into the main program, and avoid macros in the header file.

Reply:
> We can't export any widget (or some other GUI thing) from some base header. 

Why?

Reply:
This is the structure of the project!
*Base* should not be dependent on the GUI. It should compile with `nogui=true` without any additional conditionals.
Predicting your causticity about this, I note at once that there are still a few legacy dependencies from GUI (in `preferences.cpp` and `misc.cpp`), which have not yet reached the hands (apparently I need to move them to the right place soon).

Reply:
Why then, if a structure is so important, this chain of #idefss can not find its place in `src/gui/utils` directory?